src/indent.ts

```diff
@@ -29,11 +31,14 @@ export function newlineAndIndent(
         if (textEditor.document.languageId === 'python') {
             const lines = textEditor.document.getText(
                 new vscode.Range(0, 0, position.line, position.character)).split("\n");
-            let indent = nextIndentationLevel(lines, tabSize);
+
+            const { nextIndentationLevel, parseOutput: { dedentNext } } = indentationInfo(lines, tabSize);
+            let indent = nextIndentationLevel;
+
             const spacesToRemove = currentLineDedentation(lines, tabSize);
             if (spacesToRemove > 0) {
                 // don't dedent the current line if we already dedented it, e.g. after a "return"
-                if (!parseLines(lines.slice(0, -1)).dedentNext) {
+                if (!dedentNext) {
                     edit.delete(new vscode.Range(position.line, 0, position.line, spacesToRemove));
                     indent = Math.max(indent - spacesToRemove, 0);
                 }
```

Comment:
So after more investigation, I added this condition in a while back, but it's no longer needed because of a better implementation of `currentLineDedentation`.

Could you update these lines to just

```typescript
                  edit.delete(new vscode.Range(position.line, 0, position.line, spacesToRemove));
                  indent = Math.max(indent - spacesToRemove, 0);
```

(without `if (!dedentNext) {}`)

We no longer needed `dedentNext` from `indentationInfo` either

Reply:
Just pushed a commit to resolve this.